### PR TITLE
Bug 1822396: Update metric when Subscription is updated

### PR DESCRIPTION
**Pull request by anik120**
**Description of the change:**

In #1519, the subscription_sync_total metric was updated when the
related subscription object was updated. However, the PR had a bug,
in which the map used to store metric related information was not
being updated when the metric was udpated. As a result of which any
update after the first update was not removing the old metric. This
PR fixes that bug.

**Motivation for the change:**

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage 
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/docs` 
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

File: test/e2e/metrics_e2e_test.go [hunk 1]
```diff
@@ -170,17 +170,53 @@ var _ = Describe("Metrics are generated for OLM managed resources", func() {
 				}).Should(And(
 					Not(ContainElement(LikeMetric(
 						WithFamily("subscription_sync_total"),
-						WithLabel("name", "metric-subscription-for-update"),
-						WithLabel("channel", stableChannel),
+						WithName("metric-subscription-for-update"),
+						WithChannel(stableChannel),
```
Comment (benluddy) [inline test/e2e/metrics_e2e_test.go:174]: Should this matcher (and the new ones added in this PR) also specify the package?

File: test/e2e/metrics_e2e_test.go [hunk 2]
```diff
@@ -170,17 +170,56 @@ var _ = Describe("Metrics are generated for OLM managed resources", func() {
...
+							WithName("metric-subscription-for-update"),
+							WithChannel(betaChannel),
+							WithPackage(testPackageName),
+						))),
+						ContainElement(LikeMetric(
+							WithFamily("subscription_sync_total"),
+							WithName("metric-subscription-for-update"),
+							WithChannel(alphaChannel),
+							WithPackage(testPackageName),
+							WithPackage(testPackageName),
```
Comment (benluddy) [inline test/e2e/metrics_e2e_test.go:219]: It's harmless, but this line is duplicated.

File: pkg/metrics/metrics.go [hunk 3]
```diff
@@ -267,5 +267,7 @@ func UpdateSubsSyncCounterStorage(sub *olmv1alpha1.Subscription) {
 		counterValues.installedCSV = sub.Status.InstalledCSV
 		counterValues.pkg = sub.Spec.Package
 		counterValues.channel = sub.Spec.Channel
+
+		subscriptionSyncCounters[sub.GetName()] = counterValues
```
Comment (benluddy) [inline pkg/metrics/metrics.go:271]: I don't think it needs to block this bugfix, but there should really be some kind of synchronization around access to `subscriptionSyncCounters`. Would you be interested in making a separate PR with that change?

Comment (anik120) [reply pkg/metrics/metrics.go:271]: @benluddy sure! I'm curious to see what you mean by synchronization, once we discuss that I'd be happy to open up a PR if needed.
